### ISSUE TITLE
New Rule: limit-re-export

### ISSUE THREAD

**Issue by onlywei** [eslint/eslint#17787]
### Rule details

Prevent re-exporting from multiple modules

### Related ECMAScript feature

export

### What type of rule is this?

Warns about a potential problem

### Example code

```js
// bad
export a from './a';
export b from './b';

// also bad
export const c = 'c';
export d from './d';

// good
export oneThing from './e';

// also good
export const f = 'f';
export const g = 'this is the same file';
```

### Why should this rule be in the core instead of a plugin?

This is a core JavaScript feature. It doesn't have anything to do with any third party libraries.

### Participation

- [X] I am willing to submit a pull request to implement this rule.

### Additional comments

I want to be able to limit the use of barrel files: https://marvinh.dev/blog/speeding-up-javascript-ecosystem-part-7/

**Comment by fasttime** [comment]
Hi @onlywei, thanks for the issue. Could you please explain why "re-exporting from multiple modules" is a problem and clarify how it exactly occurs, so we can better understand your proposal?

I'm a little confused by those examples:

> ```js
> // bad
> export a from './a';
> export b from './b';
> 
> // good
> export oneThing from './e';
> ```

These lines look very similar to me. Why is the first block _bad_ and the other one is _good_?

**Comment by fasttime** [comment]
No reply, so closing.
